### PR TITLE
Adding support for all official sample checklists on ENA

### LINKED ISSUE #20: Adding support for all official ENA checklists
Checklists from https://www.ebi.ac.uk/ena/browser/checklists

### LINKED ISSUE #22: Using the labels from ENA in the tsv table headers instead of the underscore version
This way no mapping table is needed

### PR DESCRIPTION
**Pull request by bedroesb**
- [x] Converting all XMLs from [ENA](https://www.ebi.ac.uk/ena/browser/checklists) to template XMLs (done using a [custom script](https://github.com/bedroesb/ENA-XML-converter))
- [x] Adapt script
- [x] Update examples
- [x] Update documentation
- [x] More testing to make sure everything is preserved
- [x] Github action to keep checklists up to date
- [x] Automatic checking if there are new checklists trough API

This will close #20 and close #22

NOTE: This update will break current implementations: with this update the headers in the TSV are the same as labels by ENA (so no underscores anymore if not needed!). This means that there is no mapping needed + easier to maintain

### CODE REVIEW

**Comment by bgruening** [comment]
Interesting change :)
Should we add a GitHub action to update the checklists regularly?

**Comment by bedroesb** [comment]
haha "interesting change" sounds like a polite way to say, I would not have done that :wink: I am not convinced by putting them all in the directory here. Positive thing is that it will always work even if ENA changes their URL. I could indeed make a github action, I would need an API call where I can check which are the available checklists by ID.

**Comment by bgruening** [comment]
> haha "interesting change" sounds like a polite way to say, I would not have done that wink I am not convinced by putting them all in the directory here. Positive thing is that it will always work even if ENA changes their URL. I could indeed make a github action, I would need an API call where I can check which are the available checklists by ID.

Lol, no I think that is correct. Let me rephrase! Nice work! :)
We could download the files as you suggested on the fly, but for me this sounds brittle and especially for this case people might have restricted internet access? Maybe only upload? No clue. I just have the feeling that those files belong into the python package.

**Comment by bedroesb** [comment]
ok than it was funnier in my head than it was :D I will contact ENA to check if they can give me some details in how I can best check for changes in the checklists (especially if new ones get added)

**Comment by bedroesb** [comment]
@bgruening I added a github action that checks every Monday for changes and will make a PR if so. It makes use of a script I made in the var directory that generates the templates using jinja2. Only limitation at the moment is that I do not detect if there is a new checklist since I loop over this hardcoded list:

```
checklists = ["ERC000012", "ERC000013", "ERC000014", "ERC000015", "ERC000016", "ERC000017", "ERC000018", "ERC000019", "ERC000020", "ERC000021", "ERC000022", "ERC000023", "ERC000024", "ERC000025", "ERC000027", "ERC000028", "ERC000029", "ERC000030", "ERC000031", "ERC000032", "ERC000033", "ERC000034", "ERC000035", "ERC000036", "ERC000037", "ERC000038", "ERC000039", "ERC000040", "ERC000041", "ERC000043", "ERC000044", "ERC000045", "ERC000047", "ERC000048", "ERC000049", "ERC000050", "ERC000051", "ERC000052", "ERC000053", "ERC000011"]
```
I hope to replace this with an API call in the future

**Comment by bedroesb** [comment]
@bgruening Github action is in place + I make use of the API to detect the supported checklists

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
That is pretty damn good work! Thanks @bedroesb!
